### DISCUSSION TITLE
BPA rule to find specific expression in table with Incremental Refresh enabled

### DISCUSSION THREAD

**Discussion by edhans** [TabularEditor/TabularEditor3#998] [🙋♀️TE3 Q&A] [answered]
I have the following BPA rule that looks for our dataflow test environment and gives a warning that it needs to be fixed before going to PROD.

![image](https://github.com/TabularEditor/TabularEditor3/assets/45298358/a40ad29b-1a67-44ed-a146-63b21990f589)

This works fine in shared expressions and in the partitions of normal tables. However, for tables with IR enabled, it doesn't find it. I think it is because the M code isn't at the partition label. I tried adding "Tables" to the applies to area, but that didn't work. Suggestions?

**Comment by otykier** [comment, marked as answer]
The M expression that define incremental refresh partitions is stored in the `SourceExpression` property of a table. You may also want to check the `PollingExpression`. As such, I think you will need a separate rule that applies to Tables only, with the following rule expression:

```
SourceExpression.IndexOf("…", StringComparison.OrdinalIgnoreCase) >= 0 or
PollingExpression.IndexOf("…", StringComparison.OrdinalIgnoreCase) >= 0
```

**Reply by edhans** [reply]
Thanks. I didn't need the PollingExpression.* - worked perfectly.

I wish there was a way for the rules to allow multiple items to be added to them (Expression*, PollingExpression.*, etc) without erroring out and not working if they don't apply to a specific object (table, shared expression, etc.)

My TODO rule became 5 TODO rules to cover most of the basis where I wanted to check for that text.
